### PR TITLE
[RLlib] Curiosity minor fixes, do-overs, and testing.

### PR DESCRIPTION
**Pull request by sven1977**
This PR:
- Cleans up a little the Curiosity Exploration module.
- Fixes some bugs in it and makes sure the hyperparameters from the paper are being used as defaults.
- Adds image-based observation support to the Curiosity module.
- Adds a test case using the gym_minigrid Envs.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/evaluation/rollout_worker.py [hunk 1]
```diff
@@ -420,7 +420,7 @@ def make_env(vector_index):
         if (ray.is_initialized()
                 and ray.worker._mode() != ray.worker.LOCAL_MODE):
             # Check available number of GPUs
-            if not ray.get_gpu_ids(as_str=True):
+            if not ray.get_gpu_ids():
```
Comment (ericl) [inline rllib/evaluation/rollout_worker.py:423]: I believe as_str=True is correct to avoid crashes in certain cases, reading get_gpu_ids(). Please avoid making unrelated changes in PRs.

File: rllib/evaluation/sample_batch_builder.py [hunk 2]
```diff
@@ -196,13 +196,14 @@ def postprocess_batch_so_far(
...
                     "from a single trajectory.", pre_batch)
-            post_batches[agent_id] = policy.postprocess_trajectory(
-                pre_batch, other_batches, episode)
             # Call the Policy's Exploration's postprocess method.
+            post_batches[agent_id] = pre_batch
             if getattr(policy, "exploration", None) is not None:
                 policy.exploration.postprocess_trajectory(
                     policy, post_batches[agent_id],
                     getattr(policy, "_sess", None))
+            post_batches[agent_id] = policy.postprocess_trajectory(
```
Comment (ericl) [inline rllib/evaluation/sample_batch_builder.py:205]: Why this change?

File: rllib/models/torch/visionnet.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 from ray.rllib.utils.annotations import override
 from ray.rllib.utils.framework import get_activation_fn, try_import_torch
 
-_, nn = try_import_torch()
+torch, nn = try_import_torch()
```
Comment (ericl) [inline rllib/models/torch/visionnet.py:10]: Isn't this unused?

**Review by ericl — APPROVED** [review]
Just a couple things to fix before merging.

File: rllib/evaluation/rollout_worker.py (hunk 1 above)
Comment (sven1977) [reply rllib/evaluation/rollout_worker.py:423]: Done. It did fail in one of my runs, though. Shouldn't matter here anyways, we just check for an empty list.

File: rllib/evaluation/sample_batch_builder.py (hunk 2 above)
Comment (sven1977) [reply rllib/evaluation/sample_batch_builder.py:205]: Good question. Curiosity has to change the rewards before e.g. PPO calculates the advantages from those (otherwise, the intrinsic rewards would have no effect on the loss).
`Exploration.postprocess_trajectory()` is only used so far inside the ParameterNoise exploration class, where it doesn't matter, whether we call it before or after `Policy.postprocess_trajectory()`.
Maybe we should have two methods: before and after.

File: rllib/models/torch/visionnet.py (hunk 3 above)
Comment (sven1977) [reply rllib/models/torch/visionnet.py]: indeed. removed
